### PR TITLE
Update kitchen tests to use latest official test-kitchen

### PR DESCRIPTION
**Pull request by jdmundrawala**
Update kitchen tests to use latest official test-kitchen

cc @chef/client-core @schisamo @mcquin

### CODE REVIEW

**Comment by jdmundrawala** [comment]
This seems to be an issue with the latest test-kitchen and kitchen-ec2
cc @fnichol @tyler-ball

**Comment by tyler-ball** [comment]
Can you try this with [my kitchen-ec2 branch](https://github.com/test-kitchen/kitchen-ec2/pull/109)?  If that works for you, I'll merge 109 and you can depend on master for now until I get 0.9 released.  I need to write unit tests and update the linters, then it will be ready for release.

**Comment by tyler-ball** [comment]
@jdmundrawala I updated the yaml with configs that work for me locally.  The last travis error I saw looked like an issue reading the IAM instance profile name out of the environment variables.  Ping me tomorrow and I'll take a look at it with you.

**Comment by danielsdeleo** [comment]
Looks awesome :+1:

**Comment by tyler-ball** [comment]
:+1: @jdmundrawala you can remove the `echo` and squash all these commits

File: kitchen-tests/.kitchen.travis.yml [hunk 1]
```diff
@@ -1,35 +1,37 @@
...
   - name: centos-6.4
-    driver_plugin: ec2
-    driver_config:
-      region: "us-west-2"
-      availability_zone: "us-west-2a"
-      ssh_key: <%= ENV['EC2_SSH_KEY_PATH'] %>
-      security_group_ids: ["travis-ci"]
+    driver:
+      # https://aws.amazon.com/marketplace/seller-profile/ref=srh_res_product_vendor?id=16cb8b03-256e-4dde-8f34-1b0f377efe89
+      image_id: ami-b3bf2f83
```
Comment (schisamo) [inline kitchen-tests/.kitchen.travis.yml]: This AMI should come from the official Canonical list at:
http://uec-images.ubuntu.com/releases/precise/release/

I believe we want the 64-bit instance root store AMI: `ami-c15379f1`.

Comment (tyler-ball) [inline kitchen-tests/.kitchen.travis.yml]: This is Centos though :confused: Although if you had the official list for that, I would appreciate it!  Is http://cloud-images.ubuntu.com/locator/ec2/ a bad resource to find Ubuntu images?

Comment (schisamo) [inline kitchen-tests/.kitchen.travis.yml]: @tyler-ball totally missed that this was CentOS. In that case we should just use the RHEL AMIs we use in CI. Take a look here for a list:
https://github.com/chef/releng-chef-repo/blob/master/script/ci#L93-L96

Also `uec-images.ubuntu.com == cloud-images.ubuntu.com`.

Comment (tyler-ball) [inline kitchen-tests/.kitchen.travis.yml]: @schisamo What is the preference for using `instance` storage over `ebs`?  Is it cheaper?

Comment (btm) [inline kitchen-tests/.kitchen.travis.yml]: My memory on this was vague, but I remembered recently getting an error when trying to use an m1.small for something random and seeing the instance types had changes. Something's changing related to instance storage over at AWS, but I don't see any clear statements about it. The "m1" instances are being replaced by "t2" instances which don't support instance storage per [this page](http://aws.amazon.com/ec2/previous-generation/). Other instances still have instance storage but they're calling it "SSD" or "HDD" now on [the pricing page](http://aws.amazon.com/ec2/pricing/.

It's always seemed to me the value of instance storage over ebs storage is that you pay extra for how much ebs storage you used, but given that you don't have a choice in the small instances anymore, I guess you have to compare the cost of a t1 to an m3+ebs?

**Comment by tyler-ball** [comment]
I updated Centos to RHEL using the RelEng AMI, and updated Ubuntu to use the instance store rather than EBS.  If Travis passes, this should be :ship:able

**Comment by jdmundrawala** [comment]
hmm, I'm not sure this is correct... it seems be running the wrong provisioner :(

**Comment by jdmundrawala** [comment]
Maybe we should add an audit to the cookbook?

**Comment by tyler-ball** [comment]
@jdmundrawala I think adding an audit would be nice (it could even validate that 2+2=4)

What do you mean its running the wrong provisioner?

**Comment by tyler-ball** [comment]
Also!  Test Kitchen and kitchen-vagrant have been released - lets use those latest versions!

**Comment by jdmundrawala** [comment]
It was using chef_zero instead of chef_github

**Comment by btm** [comment]
@jdmundrawala:

```
Resolving dependencies...
Bundler could not find compatible versions for gem "test-kitchen":
  In Gemfile:
    kitchen-vagrant (~> 0.17.0.beta.2) ruby depends on
      test-kitchen (~> 1.4) ruby
    kitchen-ec2 (>= 0) ruby depends on
      test-kitchen (= 1.4.0.rc.1) ruby
    test-kitchen (~> 1.4.0.rc.1) ruby
The command "eval bundle install --jobs=3 --retry=3" failed 3 times.
The command "bundle install --jobs=3 --retry=3" failed and exited with 6 during .
Your build has been stopped.
```

**Comment by jdmundrawala** [comment]
@btm https://github.com/test-kitchen/kitchen-ec2/pull/112

**Pull request merged**